File: iosApp/Views/Onboarding/OnboardingFlow.swift
import SwiftUI

/// Top-level onboarding coordinator (spec §10).
///
/// Reads `OnboardingState.phase` and routes to the matching sub-flow:
/// admin provisioning (Phase 1), microscopist claim (Phase 2), or — when
/// the phase is `.complete` — renders nothing so the composition root
/// shows `RootView` instead.
///
/// The composition root (`MalariaDetectorApp`) decides which view to
/// mount based on the same `phase`; this view exists so that decision
/// has a single place to grow (e.g. when a `transferring` phase is
/// added for spec §10 mid-deployment transfer).
struct OnboardingFlow: View {

    @Environment(OnboardingState.self) private var onboarding

    var body: some View {
        switch onboarding.phase {
        case .adminProvisioning:
            AdminWizardView()
        case .microscopistClaim:
            MicroscopistWizardView()
        case .complete:
            // Belt-and-braces: the composition root should never mount
            // OnboardingFlow when phase is `.complete`, but if a future
            // refactor regresses that gating an empty view keeps the
            // app navigable rather than crashing.
            EmptyView()
        }
    }
}

File: androidApp/src/main/kotlin/com/malaria/android/ui/onboarding/OnboardingFlow.kt
package com.malaria.android.ui.onboarding

import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.malaria.android.services.OnboardingState
import com.malaria.android.ui.locals.LocalOnboardingState

/**
 * Top-level onboarding coordinator (spec §10).
 *
 * Reads [OnboardingState.phase] and routes to the matching sub-flow:
 * admin provisioning (Phase 1), microscopist claim (Phase 2), or — when
 * phase is [OnboardingState.Phase.Complete] — renders nothing so the
 * composition root shows `RootScreen` instead.
 *
 * Mirrors `iosApp/Views/Onboarding/OnboardingFlow.swift`.
 */
@Composable
fun OnboardingFlow() {
    val onboarding = LocalOnboardingState.current
    val phase by onboarding.phase.collectAsStateWithLifecycle()

    when (phase) {
        OnboardingState.Phase.AdminProvisioning -> AdminWizard()
        OnboardingState.Phase.MicroscopistClaim -> MicroscopistWizard()
        OnboardingState.Phase.Complete -> {
            // Belt-and-braces: the composition root should never mount
            // OnboardingFlow when phase is Complete, but an empty branch
            // keeps the app navigable rather than crashing on a future
            // gating regression.
        }
    }
}
